### DISCUSSION TITLE
Cannot  reach database server (CPANEL MySQL) when migrating. (FIRST TIME USER)

### DISCUSSION THREAD

**Discussion by Rise-admin1** [prisma/prisma#23076] [Q&A] [answered]
### Question

Hey, this is my first time using prisma and trying to access the MySQL DB provided by CPANEL. I followed all instructions from the PRISMA Docs and this is where I am currently.

prisma/schema.prisma
```prisma
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}

model Post {
  id        Int      @id @default(autoincrement())
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  title     String   @db.VarChar(255)
  content   String?
  published Boolean  @default(false)
  author    User     @relation(fields: [authorId], references: [id])
  authorId  Int
}

model Profile {
  id     Int     @id @default(autoincrement())
  bio    String?
  user   User    @relation(fields: [userId], references: [id])
  userId Int     @unique
}

model User {
  id      Int      @id @default(autoincrement())
  email   String   @unique
  name    String?
  posts   Post[]
  profile Profile?
}
```
My .env file 
```
DATABASE_URL="mysql://userName:userPassword@localhost:3306/prowrit9_prisma_journalDB"
```
When I try to migrate using ``` npx prisma migrate dev init ``` it says 
``` 
Environment variables loaded from .env Prisma schema loaded from prisma\schema.prisma Datasource "db": MySQL database "prowrit9_prisma_journalDB" at "localhost:3306"

Error: P1001: Can't reach database server at `localhost`:`3306`

Please make sure your database server is running at `localhost`:`3306`.
```

I can confirm the hostname is correct because when I navigate to the **CPANEL -> phpMyAdmin** 
![prisma](https://github.com/prisma/prisma/assets/155880624/0676602d-1cb8-4a2e-b357-89faea4a66e2)

To my knowledge, the image says the hostname and that is how I cam to the conclusion of my hostname.
 

this is my package.json 
```
{
  "name": "api",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "prisma": "^5.9.1"
  }
}

```
Is there something that I am missing? Thank You in advance.

**Comment by ludralph** [comment]
Hi @Rise-admin1 👋 

Thank you for raising this question.

Can you please try removing the quotes from your database connection string in the `.env` file? 

Instead of `DATABASE_URL="mysql://userName:userPassword@localhost:3306/prowrit9_prisma_journalDB"`, you should do this instead
`DATABASE_URL=mysql://userName:userPassword@localhost:3306/prowrit9_prisma_journalDB`. Also, if you have a special symbol like `$` sign in your password, it should be percent-encoded. See this section of the [documentation](https://www.prisma.io/docs/orm/reference/connection-urls#special-characters).

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by Rise-admin1** [reply]
Hey @ludralph, I tried removing the quotes to no avail. I think the issue is not of PRISMAS but the DB host. The DB is hosted in CPANEL localhost while I am running PRISMA from my local PC. I think If I upload my dev build to cpanel, it might connect. I am not sure myself, new to all this. Thank You for replying.

**Comment by Rise-admin1** [comment, marked as answer]
@ludralph hey I fixed it. The hostname was not supposed to be localhost like I mentioned and It should have been the server name where the DB was hosted. Got it fixed. I feel dumb lol. Thank You for the help.

**Reply by ChahineBendjeddou** [reply]
Hello @Rise-admin1,

I've encountered a similar issue to the one you mentioned previously and have attempted some troubleshooting steps without success.

I tried adjusting the database connection URL in my configuration file to use the server name "uranus-shared", like so:
```DATABASE_URL='mysql://userName:password@uranus-shared/drotop94_pscale' ```,

However, I keep receiving the same error message: ```Can't reach database server at uranus-shared:3306```

I also attempted to add my local IP address to the Remote MySQL settings, but this didn't resolve the issue.

- PS: I'm also trying to connect it to my Next.js app hosted on Vercel, but I couldn't get its IP address to whitelist it.

Your assistance in resolving this matter would be greatly appreciated. If you have any insights or suggestions on how to proceed, I would be grateful for your guidance.

Thank you for your attention to this matter.

**Reply by Rise-admin1** [reply]
Hi @ChahineBendjeddou,

I haven't delved deep into Prisma since it was fixed. I think the issue might be because you missed mentioning the port number, or your hosted MySQL might be running on a different port number. You can find the correct port number for the hosted database and add it to the DATABASE_URL like this:

```DATABASE_URL='mysql://userName:password@uranus-shared:PORT_NUMBER/drotop94_pscale'```

[DATABASE_URL Format](https://www.prisma.io/docs/getting-started/setup-prisma/start-from-scratch/relational-databases/connect-your-database-typescript-mysql)

**Reply by ChahineBendjeddou** [reply]
hi @Rise-admin1,

Thanks for your suggestion.

I appreciate your input regarding the potential issue with the port number in the DATABASE_URL configuration. I attempted to add the port number, but unfortunately, I couldn't find any information regarding the specific port number. I tried using the default port number ```3306```, but it didn't resolve the issue.
 
Additionally, I considered the possibility of an IP whitelist issue and attempted to whitelist all IPs ```%.%.%.%```. However, as expected, this did not resolve the problem.

If you have any further insights or suggestions, I would greatly appreciate your assistance.

**Reply by Rise-admin1** [reply]
Hi @ChahineBendjeddou, 
Unfortunate that did not work. Could you show your schema.prisma file, Screenshots of the error and the name of the DB hosting service that you are using. I will reopen this discussion. Somebody could help you. I think the issue might be that the DB hosted port might not be the default (3306).

**Reply by 4emar** [reply]
> @ludralph hey I fixed it. The hostname was not supposed to be localhost like I mentioned and It should have been the server name where the DB was hosted. Got it fixed. I feel dumb lol. Thank You for the help.

Hi @Rise-admin1, I am having trouble doing the same thing for 2 days now. 
I have a MySQL database on cpanel, and I cannot connect it with my Next.js app, in the same format as the connection string.

mysql://admin:password@server_name:3306/database

Where did you take the server name from?
Since in Cpanel Server Information, the server name is not related to my server. Moreover when I use my domain name, it doesn't work either. 
I have the same situation as you where in php Myadmin the top says localhost:3306, and then the database is named *user_prefix*_*database-name*.

I cannot find anything on the internet that, rather than this post, which is actually helpful.

Thanks.

**Reply by Rise-admin1** [reply]
> Where did you take the server name from? Since in Cpanel Server Information, the server name is not related to my server. Moreover when I use my domain name, it doesn't work either. I have the same situation as you where in php Myadmin the top says localhost:3306, and then the database is named _user_prefix___database-name_.

Hey, sorry for the late response. 
I will try to make this easy as posible.

Go to Bluehost -> Hosting -> (Quick Links) cPanel. This will take you to cPanel.
In cPanel, under tools, there is a database tab which has a subtab called phpMyAdmin.Open it up.

On phpMyAdmin page, Click the **Variables** tab that is visible at the top which directs you to a searchbar. Type in the keyword "hostname" which displays your hostname of your Database. 

Hope this helps.

**Reply by jagdishpadeliya** [reply]
anyone able to solve the issue. I too followed above instructions but no luck.
